lib/Compiler.js

```diff
@@ -42,44 +74,43 @@ function compileSolc(embark, contractFiles, cb) {
   
   logger.info("compiling solidity contracts with command line solc...");
 
-  const allowedDirectories = contractFiles.map((contractFile) => path.dirname(path.join(process.cwd(), contractFile.path)))
-                                          .filter((x, i, a) => a.indexOf(x) == i);
+  const allowedDirectories = contractFiles.map((contractFile) => path.dirname(contractFile.path))
+                                         .filter((x, i, a) => a.indexOf(x) == i);
 
   let compiled_object = {};
   async.each(contractFiles,
-    function (file, fileCb) {
-      compileSolcContract(logger, file.filename, allowedDirectories, (err, compileString) => {
+    function (file, callback) {
+      compileSolcContract(logger, file.filename, allowedDirectories, solcConfig, (err, compileString) => {
         if (err) {
-          return fileCb(err);
+          return cb(err);
```

Comment:
You need to call callback here (what was previously fileCb), as you don't want to continue the compilation and also it could cause double callbacks.